pkgs/by-name/dr/dragonflydb/package.nix

```diff
@@ -1,109 +1,252 @@
 {
   fetchFromGitHub,
-  fetchurl,
   lib,
   stdenv,
+
+  # Dependencies
+  abseil-cpp,
+  c-ares,
+  croncpp,
   double-conversion,
-  gperftools,
-  mimalloc,
-  rapidjson,
-  liburing,
-  xxHash,
+  expected-lite,
+  fast-float,
+  flatbuffers_23,
   gbenchmark,
-  glog,
+  gperftools,
   gtest,
+  hdrhistogram_c,
+  hnswlib,
   jemalloc,
-  gcc-unwrapped,
+  liburing,
+  lz4,
+  pcre2,
+  pugixml,
+  rapidjson,
+  re2,
+  re-flex,
+  uni-algo,
+  xxHash,
+  zstd,
+
+  # Build tools
   autoconf,
   autoconf-archive,
-  automake,
+  automake116x,
+  bison,
   cmake,
+  gcc-unwrapped,
   ninja,
+
+  # Runtime dependencies
   boost,
-  libunwind,
   libtool,
+  libunwind,
   openssl,
+  zlib,
+
+  # Options
+  withAws ? true,
+  withGcp ? true,
+  withGperf ? true,
+  withPcre ? true,
+  withRe2 ? true,
+  withSearch ? true,
+  withAsan ? false,
+  withUsan ? false,
 }:
 
 let
   pname = "dragonflydb";
-  version = "0.1.0";
+  version = "1.34.2";
 
   src = fetchFromGitHub {
-    owner = pname;
+    owner = "dragonflydb";
     repo = "dragonfly";
     tag = "v${version}";
-    hash = "sha256-P6WMW/n+VezWDXGagT4B+ZYyCp8oufDV6MTrpKpLZcs=";
+    hash = "sha256-n70IB32tZDe665hVLrKC0BSSJutmYhtPJvfNa48xaqA=";
     fetchSubmodules = true;
   };
 
-  # Needed exactly 5.4.4 for patch to work
-  lua = fetchurl {
-    url = "https://github.com/lua/lua/archive/refs/tags/v5.4.4.tar.gz";
-    hash = "sha256-L/ibvqIqfIuRDWsAb1ukVZ7c9GiiVTfO35mI7ZD2tFA=";
+  aws-sdk-cpp-1-11-162 = fetchFromGitHub {
```

Comment:
We cannot use the intree one?

Reply:
No, because it patches the AWS SDK and the patch doesn't apply against the in-tree version now, much less in the future.